src/clr-angular/forms/combobox/options.ts

```diff
@@ -1,59 +1,140 @@
+      </span>
+    </div>
 
-@Component({ selector: 'clr-options', templateUrl: './options.html', host: { '[class.clr-options]': 'true' } })
-export class ClrOptions extends AbstractPopover implements OnDestroy {
-  private sub: Subscription;
+    <!-- Rendered if data set is empty -->
+    <div *ngIf="emptyOptions">
+      <span class="clr-options-empty-text">
+        No results
+      </span>
+    </div>
+
+    <!--Option Groups and Options will be projected here-->
+    <ng-content></ng-content>
+  `,
+  providers: [{ provide: LoadingListener, useExisting: ClrOptions }],
+  host: {
+    '[class.clr-options]': 'true',
+    '[attr.aria-multiselectable]': 'optionSelectionService.multiselectable',
+    '[attr.role]': '"listbox"',
+    '[id]': 'optionsId',
+  },
+})
+export class ClrOptions<T> implements AfterViewInit, LoadingListener, OnDestroy {
+  public loading = false;
+  private subscriptions: Subscription[] = [];
 
   constructor(
-    injector: Injector,
+    public optionSelectionService: OptionSelectionService<T>,
+    @Inject(IF_ACTIVE_ID) public id: number,
+    private ariaService: AriaService,
+    private el: ElementRef,
+    public commonStrings: ClrCommonStringsService,
+    private focusHandler: ComboboxFocusHandler<T>,
+    private toggleService: ClrPopoverToggleService,
     @Optional()
     @Inject(POPOVER_HOST_ANCHOR)
-    parentHost: ElementRef
+    parentHost: ElementRef,
+    @Inject(DOCUMENT) private document: any
   ) {
     if (!parentHost) {
       throw new Error('clr-options should only be used inside of a clr-combobox');
     }
-    super(injector, parentHost);
 
-    // Configure Popover
-    this.initializeSubscriptions();
-    this.configurePopover();
+    if (!this.optionsId) {
+      this.optionsId = 'clr-options-' + nbOptionsComponents++;
+    }
+  }
+
+  public searchText(input: string) {
+    return this.commonStrings.parse(this.commonStrings.keys.searching, { INPUT: input });
   }
 
   /**
-   * Configure Popover Direction and Close indicators
+   * Tests if the list of options is empty, meaning it doesn't contain any items
    */
-  private configurePopover(): void {
-    this.anchorPoint = Point.BOTTOM_LEFT;
-    this.popoverPoint = Point.LEFT_TOP;
-    this.closeOnOutsideClick = true;
+  public get emptyOptions() {
+    return !this.optionSelectionService.loading && this.items.length === 0;
   }
 
-  private initializeSubscriptions(): void {
-    // @TODO COMBOBOX: fix on combobox merge
-    // this.sub = this.toggleService.ignoredElementChange.subscribe((el: ElementRef) => {
-    //   if (el) {
-    //     this.ignoredElement = el.nativeElement;
-    //   }
-    //   this.sub.unsubscribe();
-    // });
+  @Input('id')
+  set optionsId(id: string) {
+    this.ariaService.ariaControls = id;
+    this.ariaService.ariaOwns = id;
+  }
+
+  get optionsId(): string {
+    return this.ariaService.ariaControls;
+  }
+
+  _items: QueryList<ClrOption<T>>;
+  @ContentChildren(ClrOption)
```

Comment:
I hadn't ever thought of using a QueryList like this, neat!

I think it just needs a test that the correct things get set when a a new ClrOptions component is created (and the gettor is returning the correct thing).

I would like to see that if there is an *ngIf on one of the options the QueryList rflects the correct ClrOptions. I'm not 100% sure about this as it might just be retesting the QueryList behavior so you your discresion here

Reply:
I see. Yes, we could test this. But this will break the isolation of the tests we have now and I'll need to import and add clr-option elements to the testing template.
This will complicate the testing scenario, but if you think it will bring value, we could add it?

Reply:
Added some tests with `clr-option` child.